flybywiresim/a32nx
# feat(mcdu): Progress Page Bearing/Dist To

## Summary of Changes

I wrote this a while ago but never made a PR. Rebased and tidied up now so here it goes.
Implement bearing and distance to navaid/airport on the progress page. The values will be updated every 2 seconds when a navaid is entered (RSK 4). The duplicate names page is shown when the entered ident matches multiple navaids. ~~Currently only airports, waypoints, and navaids are supported. Lat/long, PBD, or PB/PB are not supported.~~ Supports all valid input formats: places (airports, runways, navaids), lat/long, place/bearing/distance, place-bearing/place-bearing.

Also a couple of small fixes to the duplicate names page:
 - If the distance is > 9999 Nm, show 9999, per the Honeywell step 1A manual
 - Correct the number of pages available at the top right

## Screenshots (if necessary)


![image](https://user-images.githubusercontent.com/9995998/109407737-34320900-79e8-11eb-8816-de7e56cee8a4.png)
![image](https://user-images.githubusercontent.com/9995998/109407744-44e27f00-79e8-11eb-95aa-11636f8733e9.png)


## References





Honeywell Step1A manual:
![image](https://user-images.githubusercontent.com/9995998/109407620-30ea4d80-79e7-11eb-8b78-5dbeeb1b942c.png)


## Additional context



Discord username (if different from GitHub):

## Testing instructions

- Enter a variety and waypoints, navaids, and airports; nearby and further away. Check the bearing and distance is displayed correctly, and is reasonable
- Enter a navaid approx. 10-20 nm away, and fly the bearing shown. You should end up flying directly inbound to the navaid
- Enter a navid that has a duplicate ident somewhere in the world (especially 2/3 digit ones). The duplicate names page should allow you to select the correct navaid.
- Enter a runway e.g. NZWN34 or NZAA05R while sitting at the threshold, check distance is 0 or close to
- Enter the opposite end of the runway and ensure bearing is runway heading, dist is approx. runway length
- Try entering place/bearing/distance
- Try entering place-bearing/place-bearing (suggest a runway if you're struggling, e.g. NZWN16-120/NZWN34-30 for a point to the east of NZWN 16/34)
- Try entering the various formats with places/navaids that don't exist and make sure you get an appropriate scratchpad message


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
